src/Compilers/Core/Portable/DiagnosticAnalyzer/IAnalyzerAssemblyResolver.cs

```diff
@@ -2,21 +2,39 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if NET
+
 using System.Reflection;
+using System.Runtime.Loader;
 
 namespace Microsoft.CodeAnalysis
 {
     /// <summary>
-    /// Allows a host to override how assembly resolution is performed by the <see cref="AnalyzerAssemblyLoader"/>.
+    /// This interface allows hosts to control exactly how a given <see cref="AssemblyName"/> is resolved to an
+    /// <see cref="Assembly"/> instance. This is useful for hosts that need to load assemblies in a custom way like
+    /// Razor or stream based loading.
```

Comment:
What wasn't obvious to me at first: this is a .NET Core only interface. Which makes sense given it's got the ALC, but that might be worth highlighting in the summary if you miss the #if at the start.